src/reducers/liquidity_by_token_pair/minswap.rs

```diff
@@ -0,0 +1,48 @@
+use super::model::TokenPair;
+
+pub type MinSwapPoolDatum = TokenPair;
+
+#[cfg(test)]
+mod test {
+    use crate::reducers::liquidity_by_token_pair::minswap::MinSwapPoolDatum;
+    use crate::reducers::liquidity_by_token_pair::model::PoolAsset;
+    use crate::reducers::liquidity_by_token_pair::utils::pool_asset_from;
+
+    use pallas::ledger::primitives::babbage::PlutusData;
+    use pallas::ledger::primitives::Fragment;
+
+    #[test]
+    fn test_decoding_pool_datum_ada_min() {
+        let hex_pool_datum = "d8799fd8799f4040ffd8799f581c29d222ce763455e3d7a09a665ce554f00ac89d2e99a1a83d267170c6434d494eff1b00004ce6fb73282200d87a80ff";
```

Comment:
If the pools update the scripts that involve different datum formats; we'll need to update these. 

Maybe not for now, as i don't think these will change often, but maybe it's something we can think about for future iterations.

Reply:
This would imply an upgrade of the contract, since the datum types are part of the immutable on-chain validator.
But you're correct, in that case, we need to add support for that.